addons/im_livechat/static/src/legacy/widgets/feedback.js

```diff
@@ -0,0 +1,156 @@
+     * @param {Object} options
+     */
+    _sendFeedback: function (reason) {
+        var self = this;
+        var args = {
+            uuid: this._livechat.getUUID(),
+            rate: this.rating,
+            reason: reason,
+        };
+        this.dp.add(session.rpc('/im_livechat/feedback', args)).then(function (response) {
+            var emoji = RATING_TO_EMOJI[self.rating] || "??";
+            if (!reason) {
+                var content = _.str.sprintf(_t("Rating: %s"), emoji);
+            }
+            else {
+                var content = "Rating reason: \n" + reason;
+            }
+            self.trigger('send_message', { content: content, isFeedback: true });
+        });
+    },
+    /**
+    * @private
+    */
+    _showThanksMessage: function () {
+        this.$('.o_livechat_rating_box').empty().append($('<div />', {
+            text: _t('Thank you for your feedback'),
+            class: 'text-muted'
+        }));
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onClickNoFeedback: function () {
+        this.trigger('feedback_sent'); // will close the chat
+    },
+    /**
+     * @private
+     */
+    _onClickSend: function () {
+        this.$('.o_livechat_rating_reason').hide();
+        this._showThanksMessage();
+        if (_.isNumber(this.rating)) {
+            this._sendFeedback(this.$('textarea').val());
+        }
+    },
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickSmiley: function (ev) {
+        this.rating = parseInt($(ev.currentTarget).data('value'));
+        this.$('.o_livechat_rating_choices img').removeClass('selected');
+        this.$('.o_livechat_rating_choices img[data-value="' + this.rating + '"]').addClass('selected');
+
+        // only display textearea if bad smiley selected
+        if (this.rating !== 5) {
+            this._sendFeedback();
+            this.$('.o_livechat_rating_reason').show();
+        } else {
+            this.$('.o_livechat_rating_reason').hide();
+            this._showThanksMessage();
+            this._sendFeedback();
+        }
+    },
+    /**
+    * @private
+    */
+    _onEmailChat: function () {
+        var self = this;
+        var $email = this.$('#o_email');
+
+        if (utils.is_email($email.val())) {
+            $email.removeAttr('title').removeClass('is-invalid').prop('disabled', true);
+            this.$('.o_email_chat_button').prop('disabled', true);
+            this._rpc({
+                route: '/im_livechat/email_livechat_transcript',
+                params: {
+                    uuid: this._livechat.getUUID(),
+                    email: $email.val(),
+                }
+            }).then(function () {
+                self.$('.o_livechat_email').html($('<strong />', { text: _t('Conversation Sent') }));
```

Comment:
@seb-odoo move of this like of code that make use of `html()` (trigger `ci/security`).
Code looks very safe.

Reply:
Don't need to overwrite ci/security in this PR: we'll make a grouped PR, so we'll need this procedure in the grouped PR.